src/frontend/screens/AppPasscode/index.tsx

```diff
@@ -24,14 +24,20 @@ export type PasscodeScreens =
   | "enterPasscode"
   | "disablePasscode";
 
-export const AppPasscode: NativeNavigationComponent<"AppPasscode"> = () => {
-  const { authValuesSet: authenticationValuesSet } = React.useContext(
-    SecurityContext
-  );
+export const AppPasscode: NativeNavigationComponent<"AppPasscode"> = ({
+  navigation,
+}) => {
+  const { authValuesSet, obscureModeOn } = React.useContext(SecurityContext);
   const [screenState, setScreenState] = React.useState<PasscodeScreens>(() =>
-    authenticationValuesSet.passcodeSet ? "enterPasscode" : "intro"
+    authValuesSet.passcodeSet ? "enterPasscode" : "intro"
   );
 
+  React.useEffect(() => {
+    if (obscureModeOn) {
+      navigation.navigate("Settings");
+    }
+  }, [navigation, obscureModeOn]);
+
```

Comment:
What is this doing? As I understand it, if the user navigates here from the settings screen, and obscure mode is on, then they will immediately be navigated back? Or is this to catch when the user returns to the app from being on one of these screens and they turn the obscure mode on when returning? It would be good to document.

Reply:
If the user is in obscure mode, we don't want any of the security features to be visible. In the setting, the security features are hidden when in obscure mode, so this just guarantees that the user does not navigate to those screen